### DISCUSSION TITLE
Is it possible to get the current active user in middleware?

### DISCUSSION THREAD

**Discussion by hyperleex** [fastapi/fastapi#7787] [Questions] [answered]
I need to register information about the user and the body of the response, in Django it is easy to implement it in the middleware.
Is it possible to get the current active user in middleware?
```python
data = {
    "flow_id": request.flow_id,
    "request": {
        "method": request.method,
        "uri": request.get_raw_uri(),
        "login": str(getattr(request.user, 'login', "")) or None,
        "data": json.dumps(rest_request.data, ensure_ascii=False) if rest_request.data else None,
        "headers": self._join_headers(request.headers._store.values())
    },
    "response": {
        "status": rest_response.status_code,
        "reason": rest_response.reason_phrase,
        "data": json.dumps(
            rest_response.data, ensure_ascii=False) if rest_response.data is not None else None,
        "headers": self._join_headers(response._headers.values())
    },
    "elapsed": elapsed
}
```

**Comment by hyperleex** [comment]
Or Is it possible to get the current active user in custom APIRouter?
and if i set` route_class=TimedRoute` to parent router, it is not work on child router `games.router`
``` python
api_router = APIRouter(route_class=TimedRoute)

api_router.include_router(
    games.router,
    prefix="/games",
)
```

**Comment by tiangolo** [comment]
I assume you solved your problem, so thanks for reporting back and closing the issue :+1:

**Comment by ariminia** [comment]
Hey, Could you please share the solution you found? I need to implement a custom router that logs user and api url to obtain metrics on api usage per user.

**Comment by ariminia** [comment]
hi @hyperleex, Can you please share the solution here.

**Reply by AthreyVinay** [reply]
@ari-jain013 Did you have any luck in finding the solution? Thanks

**Reply by ariminia** [reply, marked as answer]
Hey I was able to store data in request.state.<variable_name> and then fetch it as needed. Apologies for the delay. Feel free to post any better alternatives.
